### ISSUE TITLE
orderby before join clause gets "swallowed" when query also has a collection property

### ISSUE THREAD

**Issue by ykosbie** [aspnet/EntityFramework6#354]
Possibly related to #353.  (Also a problem w/ orderby clause position affecting SQL output).

**tl;dr;** A `orderby contact.Property` clause before a join (any type) dissapears from the final SQL if the select clause includes a collection property.

Question

I have a simple data model for storing contact information

    public class Contact {
        public int ContactID { get; set; }

        public virtual ICollection<Address> Addresses { get; set; }
        //etc
    }

    public class Address {
        public int AddressID { get; set; }

        public int ContactID { get; set; }
        public virtual Contact Contact { get; set; }
        //etc
    }

I want to project that into a flat model and sort the results, for example

	from c in Contacts
	join a in Addresses on c.ContactID equals a.AddressID
	orderby c.DOB
	select new {
		ContactID = c.ContactID,

		CurrentAddress = a,
		AllAddresses = c.Addresses
	}

That generates SQL with an `ORDER BY [DOB]` as expected.
	
	SELECT 
	    [Project1].[ContactID] AS [ContactID], 
	    [Project1].[AddressID] AS [AddressID], 
	    [Project1].[C1] AS [C1], 
	    [Project1].[AddressID1] AS [AddressID1]
	    FROM ( SELECT 
	        [Extent1].[ContactID] AS [ContactID], 
	        [Extent1].[DOB] AS [DOB], 
	        [Extent2].[AddressID] AS [AddressID], 
	        [Extent3].[AddressID] AS [AddressID1], 
	        CASE WHEN ([Extent3].[AddressID] IS NULL) THEN CAST(NULL AS int) ELSE 1 END AS [C1]
	        FROM   [dbo].[Contacts] AS [Extent1]
	        INNER JOIN [dbo].[Addresses] AS [Extent2] ON [Extent1].[ContactID] = [Extent2].[ContactID]
	        LEFT OUTER JOIN [dbo].[Addresses] AS [Extent3] ON [Extent1].[ContactID] = [Extent3].[ContactID]
	    )  AS [Project1]
	    ORDER BY [Project1].[DOB] ASC, [Project1].[ContactID] ASC, [Project1].[AddressID] ASC, [Project1].[C1] ASC

If I switch the order of the `orderby c.DOB` to be before `join a in Addresses...`, EF *completely removes the ORDER BY* from the generated SQL!

	SELECT 
	    [Project1].[ContactID] AS [ContactID], 
	    [Project1].[AddressID] AS [AddressID], 
	    [Project1].[C1] AS [C1], 
	    [Project1].[AddressID1] AS [AddressID1]
	    FROM ( SELECT 
	        [Extent1].[AddressID] AS [AddressID], 
	        [Extent1].[ContactID] AS [ContactID], 
	        [Extent2].[AddressID] AS [AddressID1], 
	        CASE WHEN ([Extent2].[AddressID] IS NULL) THEN CAST(NULL AS int) ELSE 1 END AS [C1]
	        FROM  [dbo].[Addresses] AS [Extent1]
	        LEFT OUTER JOIN [dbo].[Addresses] AS [Extent2] ON [Extent1].[ContactID] = [Extent2].[ContactID]
	    )  AS [Project1]
	    ORDER BY [Project1].[ContactID] ASC, [Project1].[AddressID] ASC, [Project1].[C1] ASC

**Comment by ajcvickers** [comment]
@ykosbie As per the way LINQ works in general (e.g. Linq-to-objects as an example), once a join is performed, any ordering that happened before the join is not preserved. Therefore this is by-design behavior. /cc @divega who may want to comment further.
